src/core/qgsmaptopixel.h

```diff
@@ -237,10 +237,26 @@ class CORE_EXPORT QgsMapToPixel
      * \param widthPixels Output width, in pixels
      * \param heightPixels Output height, in pixels
      * \param rotation clockwise rotation in degrees
+     *
+     * \note if parameters leads to a non valid transform, parameters are not changed
      * \since QGIS 2.8
      */
     void setParameters( double mapUnitsPerPixel, double centerX, double centerY, int widthPixels, int heightPixels, double rotation );
 
+    /**
+     * Set parameters for use in transforming coordinates
+     * \param mapUnitsPerPixel Map units per pixel
+     * \param centerX X coordinate of map center, in geographical units
+     * \param centerY Y coordinate of map center, in geographical units
+     * \param widthPixels Output width, in pixels
+     * \param heightPixels Output height, in pixels
+     * \param rotation clockwise rotation in degrees
+     * \param ok will be set to true if parameters leads to a valid transform, otherwise  parameters are not changed and ok will be set to false
```

Comment:
```suggestion
     * \param ok will be set to TRUE if the specified parameters result in a valid transform, otherwise the changes are ignored and ok will be set to FALSE.
```